CMakeLists.txt

```diff
@@ -165,6 +155,20 @@ if(WITH_STL)
   endif()
 endif()
 
+if(WITH_GSL)
+  add_definitions(-DHAVE_GSL)
```

Comment:
Just to clarify, with this PR GSL library will only be used if the `WITH_GSL` flag is enabled?

Reply:
yes it is enabled only when the user explicitly asks for it.

Reply:
Ok, I am concerned seeing this part:
https://github.com/open-telemetry/opentelemetry-cpp/blob/05d43d61ed162eea063cbd4c0ea19728d106b6b3/api/CMakeLists.txt#L48-L51

As it looks we are enabling HAVE_GSL macro with WITH_STL option, which would mean try using GSL from submodule if STL doesn't have std::span implementation.

Reply:
that is covered [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/bc1b469c72f12e068af10c73829f1486eb006a2d/api/include/opentelemetry/std/span.h#L29). We shouldn't define it though, I cleaned it from the CMake file.

Reply:
ok, thanks for the clarification.